dialogue-core/src/main/java/com/palantir/dialogue/core/TimingEndpointChannel.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Timer;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.palantir.dialogue.Endpoint;
+import com.palantir.dialogue.EndpointChannel;
+import com.palantir.dialogue.Request;
+import com.palantir.dialogue.Response;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.io.IOException;
+
+final class TimingEndpointChannel implements EndpointChannel {
+    private final EndpointChannel delegate;
+    private final Timer responseTimer;
+    private final Meter ioExceptionMeter;
+
+    private TimingEndpointChannel(EndpointChannel delegate, Timer responseTimer, Meter ioExceptionMeter) {
+        this.delegate = delegate;
+        this.responseTimer = responseTimer;
```

Comment:
I thought we weren’t using a timer anymore because it doesn’t give us specific enough (per-endpoint) data to be useful?

Reply:
Hmm I'm always hesitant to delete metrics because of how salty people got before... I guess the thing I find most interesting is actually the rate of `client.response.count`.

Reply:
I thought it was already a meter

Reply:
https://github.com/palantir/dialogue/blob/4619ce4d2ea8fb8715e21d9ddb796ca075cd4a4e/dialogue-core/src/main/metrics/dialogue-core-metrics.yml#L10

Reply:
oof, the thumbs up on this never resulted in a merge :-(
https://github.com/palantir/dialogue/pull/757

Reply:
Ah dang, I can cherry-pick your change across here if you like?

Reply:
works for me, otherwise I can fix up the conflicts ¯\_(ツ)_/¯

Reply:
Chatter over VTC, resolved to keep this as a timer as this gives us parity with c-j-r and people have established dashboards around this.